include/ColorChooser.h

```diff
@@ -21,15 +21,26 @@
  *
  */
 
-#include <QColorDialog>
 #include <QApplication>
+#include <QColor>
+#include <QColorDialog>
 #include <QKeyEvent>
+#include <QVector>
 
 class ColorChooser: public QColorDialog
 {
 public:
 	ColorChooser(const QColor &initial, QWidget *parent): QColorDialog(initial, parent) {};
 	ColorChooser(QWidget *parent): QColorDialog(parent) {};
+	
+	enum class CCPalette {Default, Track, Mixer};
+	
+	//! Set global palette via array, checking bounds
+	void setPalette (const QVector<QColor>);
+	//! Set global paletter via enum
+	void setPalette (const CCPalette);
+	//! Set palette via enum, return self pointer for chaining
+	ColorChooser* setPaletteAndPoint (const CCPalette);
```

Comment:
Maybe we can call this `withPalette` instead? Then usage becomes `myChooser.withPalette(palette)->getColor()`, which is quite nice I think.

Reply:
Alright, will rename.